private/buf/bufcli/bufcli.go

```diff
@@ -637,6 +638,36 @@ func newRegistryProviderWithOptions(container appflag.Container, opts ...bufapic
 	return bufapiclient.NewConnectClientProvider(container.Logger(), client, options...)
 }
 
+// NewConnectClientConfig creates a new connect.ClientConfig which uses a token reader to look
+// up the token in the container or in netrc based on the address of each individual client.
+// It is then set in the header of all outgoing requests from clients created using this config.
+func NewConnectClientConfig(container appflag.Container) (*connectclient.Config, error) {
+	// TODO: when the generated provider is ripped out, create this config directly
```

Comment:
We've been leaving `TODO(BSR-XXX)` to make sure we don't loose sight of cleanups like this, can we add a subticket to BSR-336?

Reply:
```suggestion
	// TODO(BSR-798): when the generated provider is ripped out, create this config directly
```